File: Solutions/하루 한문제 코테준비/javascript_solve/튜플_프로그래머스_64065.js
function solution(s) {
    var answer = [];
    const p = /{[0-9,]+}/g;

    let part = s.match(p);
    let resultPart = [];
    part.forEach(v => {
        // console.log(v)
        resultPart.push(v.slice(1,v.length-1).split(','))
    });
    resultPart.sort((a,b)=>{
        return a.length-b.length;
    })
    let result = new Map();
    resultPart.forEach(v=>{
        v.forEach(each=>{
            result.set(parseInt(each,10),1);
        });
    });
    // console.log(Array.from(result.keys()))

    return Array.from(result.keys());
}
console.log(solution("{{1,2,3},{2,1},{1,2,4,3},{2}}"))

File: Solutions/하루 한문제 코테준비/튜플_프로그래머스_64065.py
import re
def solution(s):
    answer = []
    p = re.compile('{[0-9,]+}')
    # print(p.findall(s))
    part = p.findall(s)
    resultPart = []
    for i in part:
        resultPart.append(list(map(int,i[1:-1].split(','))))
    resultPart.sort(key=lambda x:len(x))
    # print(resultPart)
    result = []
    for i in resultPart:
        for j in i:
            result.append(j)
    # print(list(dict.fromkeys(result)))
    return list(dict.fromkeys(result))

print(solution("{{2},{2,1},{2,1,3},{2,1,3,4}}"))
print(solution("{{1,2,3},{2,1},{1,2,4,3},{2}}"	))